activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/ProcessStartedActorProviderEventListener.java

```diff
@@ -51,7 +55,7 @@ public void onEvent(ProcessCreatedEvent event) {
                     .ifPresent(details ->
                         runtimeService.addUserIdentityLink(
                             event.getEntity().getId(),
-                            principalIdentityProvider.getUserId(principal),
+                            identityService.findUserByName(event.getEntity().getInitiator()).getId(),
```

Comment:
It is not a good idea to use the identity service to enrich the identity link with initiator user guid, which will make an external rest api call to identity provider inside engine transaction. It will cause a delay for every new process instance which will impact the performance of the system in a negative way. 

Besides, not every process instance will have an initiator, because it may be created by system on timer or message events. In this case, the initiator will be null and it will result in NPE.